code/modules/mob/living/simple_animal/hostile/thing.dm

```diff
@@ -0,0 +1,2046 @@
+#define THING_NOARMBLADE 0
+#define THING_TRANSITIONARMBLADE 1
+#define THING_ARMBLADE 2
+
+/mob/living/simple_animal/hostile/thing  // Originally named "thing".
+	name = "unknown"
+	desc = "What is that?"
+	icon = 'icons/mob/mob.dmi'
+	icon_state = "thing"
+	icon_living = "thing"
+	icon_dead =  "thing_dead"
+	maxHealth = 400
+	health = 400
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	environment_smash = 1
+	universal_speak = 1
+	response_help = "tries to touch the"
+	response_disarm = "attempts to shove the"
+	response_harm = "hits the"
+	attacktext = "enlightens"
+	attack_sound = "enlighten"
+	speak_emote = list("articulates", "speaks")
+	gold_core_spawnable = CHEM_MOB_SPAWN_INVALID
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 1500
+	friendly = "stares at"
+	faction = list("thing")
+	deathmessage = "dissipates!"
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+	see_in_dark = 6
+	death_sound = 'sound/thing/death.ogg'
+	var/phasing = FALSE // Phasing?
+	var/armblade = THING_NOARMBLADE // Can be THING_NOARMBLADE or THING_TRANSITIONARMBLADE or THING_ARMBLADE.
+	var/prying = FALSE // Opening a door?
+	var/absorbing = FALSE // Absorbing?
+	var/phasing_cooldown = 6
+	var/sound_cooldown = 10 // Cooldown to avoid abuse
+	var/absorb_target = null // Thing to absorb.
+	var/points = 0 // Points to spend in upgrades. Starts with 0.
+	var/absorptions = 0 // For special abilities that require you to work hard for them.
+	var/upgrading = FALSE
+//Abilities
+	var/can_travel = TRUE // Can travel? Legacy variable.
+	var/can_weaponize = FALSE // Can weaponize?
+	var/can_convert = FALSE // Can convert those who it absorbs into monsters?
+	var/can_omnivision = FALSE // Can see everything?
+	var/can_screech = FALSE // Can screech
+	var/screech_level = 1 // Level 1 or 2
+	var/screech_cooldown = 80
+	var/can_emp = FALSE // Can EMP
+	var/emp_level = 1 // Level 1 or 2
+	var/emp_cooldown = 120
+	var/can_build = FALSE // Can build things?
+	var/creep_cooldown = 35
+	var/build_cooldown = 180 // Use a different cooldown for actual buildings
+	var/can_evolve = FALSE // Can become a big, big boy?
+
+/mob/living/simple_animal/hostile/thing/Process_Spacemove(movement_dir = 0) // Can move freely in space
+	return TRUE
+
+/mob/living/simple_animal/hostile/thing/New()
+	..()
+//	world << sound('sound/thing/warning.ogg') Make this sound play with the event that spawns it. Or play it manually.
+	var/datum/action/innate/thing/travel/travel_action = new()
+	travel_action.Grant(src)
+	var/datum/action/innate/thing/absorb/absorb_action = new()
+	absorb_action.Grant(src)
+	var/datum/action/innate/thing/empower/empower_action = new()
+	empower_action.Grant(src)
+
+/mob/living/simple_animal/hostile/thing/proc/UndefinedCommunication()
+	var/message = input(src, "Announce to your spawn", "Undefined Communication")
```

Comment:
Always use stripped_input. See the guide at contributing.md. You DO NOT trust players with raw text input.